internal/datastore/crdb/watch.go

```diff
@@ -195,10 +200,237 @@ func (cds *crdbDatastore) Watch(ctx context.Context, afterRevision datastore.Rev
 				}
 				errs <- datastore.NewWatchCanceledErr()
 			} else {
-				errs <- changes.Err()
+				errs <- err
 			}
 			return
 		}
 	}()
 	return updates, errs
 }
+
+type schemaChangeDetails struct {
+	Resolved string
+	Updated  string
+	After    *struct {
+		Namespace                 string `json:"namespace"`
+		SerializedNamespaceConfig string `json:"serialized_config"`
+
+		CaveatName                 string `json:"name"`
+		SerializedCaveatDefinition string `json:"definition"`
+	}
+}
+
+const maxSchemaWatchRetryCount = 5
+
+func (cds *crdbDatastore) WatchSchema(ctx context.Context, afterRevision datastore.Revision) (<-chan *datastore.SchemaState, <-chan error) {
+	updates := make(chan *datastore.SchemaState, cds.watchBufferLength)
+	errs := make(chan error, 1)
+
+	features, err := cds.Features(ctx)
+	if err != nil {
+		errs <- err
+		return updates, errs
+	}
+
+	if !features.Watch.Enabled {
+		errs <- datastore.NewWatchDisabledErr(fmt.Sprintf("%s. See https://spicedb.dev/d/enable-watch-api-crdb", features.Watch.Reason))
+		return updates, errs
+	}
+
+	lastRevision := afterRevision
+	retryCount := -1
+	go (func() {
+		running := true
+		for running {
+			retryCount++
+
+			log.Debug().Int("sleep-seconds", retryCount).Msg("sleeping before connecting to schema watch")
+			time.Sleep(time.Duration(retryCount) * time.Second)
```

Comment:
`pool` already uses a library for this: https://github.com/authzed/spicedb/blob/main/internal/datastore/crdb/pool/pool.go#L327

Reply:
Nice! I'll reuse that